### PR TITLE
add reason code

### PR DESCRIPTION
**Pull request by jgrant22**
RXA-19 to Immunization.reasonCode

### CODE REVIEW

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7ImmunizationFHIRConversionTest.java [hunk 1]
```diff
@@ -124,12 +124,16 @@ public void testImmunization() throws IOException {
...
+                + "RXA|0|1|20130531|20130531|48^HIB PRP-T^CVX|0.5|ML^^UCUM||00^new immunization record^NIP001|^Sticker^Nurse|^^^RI2050||||33k2a|20131210|PMC^sanofi^MVX||00^refusal|CP|A\r"
                 + "RXR|C28161^IM^NCIT^IM^INTRAMUSCULAR^HL70162|RT^right thigh^HL70163\r"
                 + "OBX|1|CE|64994-7^vaccine fund pgm elig cat^LN|1|V02^VFC eligible Medicaid/MedicaidManaged Care^HL70064||||||F|||20130531|||VXC40^per imm^CDCPHINVS\r";
 
         Immunization immunization2 = ResourceUtils.getImmunization(hl7VUXmessageRep);
 
+        assertThat(immunization2.hasReasonCode());
+        assertThat(immunization2.getReasonCodeFirstRep().getCodingFirstRep().getCode()).isEqualTo("00");
+        assertThat(immunization2.getReasonCodeFirstRep().getCodingFirstRep().getDisplay()).isEqualTo("refusal");
+        assertThat(immunization2.getReasonCodeFirstRep().getCodingFirstRep().getSystem()).isNull();
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7ImmunizationFHIRConversionTest.java:136]: add a check for the CodeableConcept.text (also null)

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7ImmunizationFHIRConversionTest.java [hunk 2]
```diff
@@ -139,4 +143,5 @@ public void testImmunization() throws IOException {
     // TODO: 10/15/21 RXA-9 (also mapped to primarySource)
     //  RXA-18 statusReason
     //  RXA-20 (status, statusReason, isSubpotent)
+    // any examples of RXA-19? not seeing any in the test examples
```
Comment (LisaWellman) [inline src/test/java/io/github/linuxforhealth/hl7/segments/Hl7ImmunizationFHIRConversionTest.java:146]: I looked and also did not find examples.
It is fine to make up your own message.
Remove this comment.

**Review by LisaWellman — APPROVED** [review]
Small comments to address, approving

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7ImmunizationFHIRConversionTest.java (hunk 2 above)
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7ImmunizationFHIRConversionTest.java]: done

File: src/test/java/io/github/linuxforhealth/hl7/segments/Hl7ImmunizationFHIRConversionTest.java (hunk 1 above)
Comment (jgrant22) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7ImmunizationFHIRConversionTest.java:136]: its pulling text from CWE.2 ?

Comment (LisaWellman) [reply src/test/java/io/github/linuxforhealth/hl7/segments/Hl7ImmunizationFHIRConversionTest.java:136]: ah, yes, if it is in CWE-2 that will be used for "text" - corrected my comment

**Pull request merged**